### DISCUSSION TITLE
LD can't find libz-ng.a

### DISCUSSION THREAD

**Discussion by kbieniex** [zlib-ng/zlib-ng#1439] [Q&A] [answered]
I'm trying to compile a project, which uses zlib-ng as a third party library (not in compat mode) using CMake with:
```
cmake .. -G Ninja
cmake --build .
ninja install
```
CMake configuration:
```Cmake
set(ZLIB_ENABLE_TESTS ON)
set(ZLIB_NG_BUILD_SHARED_LIBS OFF)

set(ZLIB_NG_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src/thirdparty/zlib-ng")
set(ZLIB_NG_BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/src/thirdparty/zlib-ng")

add_subdirectory(${ZLIB_NG_SOURCE_DIR} ${ZLIB_NG_BINARY_DIR} EXCLUDE_FROM_ALL)

set(ZLIB_NG_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/zlib-ng CACHE STRING "Path to zlib include directory")
set(ZLIB_NG_LIBRARY ${CMAKE_CURRENT_BINARY_DIR}/zlib-ng CACHE STRING "Path to zlib directory")

include_directories(${ZLIB_NG_BINARY_DIR})
include_directories(${ZLIB_NG_SOURCE_DIR})

set(ZLIB_NG "libz-ng")

target_link_libraries (PROJECT_NAME ${ZLIB_NG})
```

I'm getting an error `/usr/bin/ld: cannot find -llibz-ng: No such file or directory`

What am I doing wrong?

**Comment by nmoinvaz** [comment, marked as answer]
I think target_link_libraries needs the name of the CMake project (ie zlib), instead of the name of the library output; probably due to EXCLUDE_FROM_ALL.

**Reply by kbieniex** [reply]
When I change `target_link_libraries (PROJECT_NAME ${ZLIB_NG})` to `target_link_libraries (PROJECT_NAME "zlib-ng")`, the same error apears but with zlib-ng
`/usr/bin/ld: cannot find -lzlib-ng: No such file or directory`

**Reply by kbieniex** [reply]
Ok, got it. Didn't realise the project name is just "zlib" and not "zlib-ng"
